src/plugins/data/public/search/search_service.ts

```diff
@@ -74,11 +75,16 @@ export class SearchService implements Plugin<ISearchSetup, ISearchStart> {
   private contextContainer?: IContextContainer<TSearchStrategyProvider<any>>;
 
   private search?: ISearchGeneric;
+  private readonly loadingCount$ = new BehaviorSubject(0);
 
   constructor(private initializerContext: PluginInitializerContext) {}
 
   public setup(core: CoreSetup): ISearchSetup {
-    const search = (this.search = createAppMountSearchContext(this.searchStrategies).search);
+    core.http.addLoadingCountSource(this.loadingCount$);
```

Comment:
I'm not familiar with this `addLoadingCountSource`, could you explain?

Reply:
This code registeres (and later triggers) an observable, into a service which is used to show the loading indicator on top of the application when a network operation occurs.

https://github.com/elastic/kibana/blob/master/src/core/public/chrome/chrome_service.tsx#L174

This is automatically triggered when executing a search using `es` and this behavior needed to be restored using new platform.